### PR TITLE
Added docs guide for HTML comment feature

### LINKED ISSUE #10159: HTML comments: write docs guide
## Provide a description of the task

The docs guide must be created.

### PR DESCRIPTION
**Pull request by psmyrek**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/guides/contributing/git-commit-message-convention.html))

Internal (html-support): Added docs guide for HTML comment feature. Closes #10159.

### CODE REVIEW

File: packages/ckeditor5-html-support/docs/features/general-html-support.md [hunk 1]
```diff
@@ -233,6 +235,49 @@ htmlSupport: {
 
 The above configuration will work similarly to [`allowedContent: true`](/docs/ckeditor4/latest/api/CKEDITOR_config.html#cfg-allowedContent) option from CKEditor 4.
 
+## HTML comments
+
+By default, all HTML comments are filtered out during the editor initialization. The {@link module:html-support/htmlcomment~HtmlComment} feature allows developers to keep comments in the document content and retrieve them back, i.e. during {@link builds/guides/integration/saving-data saving editor data}. Comments are transparent from the user point of view and they are not displayed in the editable element.
```
Comment (ma2ciek) [inline packages/ckeditor5-html-support/docs/features/general-html-support.md:240]:
```suggestion
By default, all HTML comments are filtered out during the editor initialization. The {@link module:html-support/htmlcomment~HtmlComment} feature allows developers to keep them in the document content and retrieve them back, e.g. during {@link builds/guides/integration/saving-data saving the editor data}. Comments are transparent from the user point of view and they are not displayed in the editable element.
```

File: packages/ckeditor5-html-support/docs/features/general-html-support.md [hunk 2]
```diff
@@ -233,6 +235,49 @@ htmlSupport: {
...
+
+Please note that the support for HTML comments is at the basic level so far - see the [known issues](#known-issues) section below.
+
+<info-box>
+	The HTML comment feature is **experimental and not yet production-ready**.
+</info-box>
```
Comment (ma2ciek) [inline packages/ckeditor5-html-support/docs/features/general-html-support.md:246]: Perhaps we could merge the above sentence into the warning box.

File: packages/ckeditor5-html-support/src/htmlcomment.js [hunk 3]
```diff
@@ -13,7 +13,7 @@ import { uid } from 'ckeditor5/src/utils';
 /**
  * The HTML comment feature.
  *
- * For a detailed overview, check the {@glink features/html-comment HTML comment feature documentation}.
+ * For a detailed overview, check the {@glink features/general-html-support#html-comments HTML comment feature documentation}.
```
Comment (ma2ciek) [inline packages/ckeditor5-html-support/src/htmlcomment.js:16]: I'd just add one sentence to the plugin API docs like `Preserves HTML comments (<-- -->) in the editor data.`

File: packages/ckeditor5-html-support/docs/features/general-html-support.md [hunk 4]
```diff
@@ -233,6 +235,49 @@ htmlSupport: {
...
+	.catch( ... );
+```
+
+<info-box info>
+	Read more about {@link builds/guides/integration/installing-plugins installing plugins}.
+</info-box>
+
+HTML comment feature does not require any configuration.
+
 ## Known issues
```
Comment (ma2ciek) [inline packages/ckeditor5-html-support/docs/features/general-html-support.md:281]: Maybe we could have one section for known issues in general HTML support and one for HTML comments?

File: packages/ckeditor5-html-support/docs/features/general-html-support.md [hunk 5]
```diff
@@ -244,6 +289,8 @@ Most of the existing CKEditor 5 features can already be extended this way, howev
 * Some of the media embed features' markup [#9918](https://github.com/ckeditor/ckeditor5/issues/9918).
 * The `<ul>` and `<ol>` elements of the list feature [#9917](https://github.com/ckeditor/ckeditor5/issues/9917).
 
+The main issue in HTML comment feature is that comments can be easily repositioned or lost in various cases [#10118](https://github.com/ckeditor/ckeditor5/issues/10118), [#10119](https://github.com/ckeditor/ckeditor5/issues/10119).
```
Comment (ma2ciek) [inline packages/ckeditor5-html-support/docs/features/general-html-support.md:292]: We have also an issue with drag & drop / copy-pasting.

File: packages/ckeditor5-html-support/src/htmlcomment.js [hunk 6]
```diff
@@ -11,7 +11,7 @@ import { Plugin } from 'ckeditor5/src/core';
 import { uid } from 'ckeditor5/src/utils';
 
 /**
- * The HTML comment feature.
+ * The HTML comment feature. It preserves HTML comments (<!-- -->) in the editor data.
```
Comment (ma2ciek) [inline packages/ckeditor5-html-support/src/htmlcomment.js:14]:
```suggestion
 * The HTML comment feature. It preserves HTML comments (`<!-- -->`) in the editor data.
```

**Comment by ma2ciek** [comment]
@godai78, could you take a look?

**Comment by ma2ciek** [comment]
IMO the HTML comment guide doesn't fit perfectly into the shape of our GHS docs, but we can extract it later on.

**Review by ma2ciek — APPROVED** [review]

**Review by godai78 — APPROVED** [review]
I think we should revisit the idea of the structure, where we have separate features in a single guide, but maybe later, when there are, like, 3 of those or smthn. So far so good.

**Pull request merged**